### PR TITLE
[full-ci] Add support for the jsoncs3 public share manager

### PR DESCRIPTION
**Pull request by aduffeck**
This PR allows for configuring the jsoncs3 public share manager from https://github.com/cs3org/reva/pull/3199

FIxes https://github.com/owncloud/ocis/issues/3881
Fixes https://github.com/owncloud/ocis/issues/2164

### CODE REVIEW

File: services/sharing/pkg/config/config.go [hunk 1]
```diff
@@ -110,8 +110,9 @@ type UserSharingJSONCS3Driver struct {
 }
 
 type PublicSharingDrivers struct {
-	JSON PublicSharingJSONDriver `yaml:"json"`
-	CS3  PublicSharingCS3Driver  `yaml:"cs3"`
+	JSON    PublicSharingJSONDriver    `yaml:"json"`
+	CS3     PublicSharingCS3Driver     `yaml:"cs3"`
+	JSONCS3 PublicSharingJSONCS3Driver `yaml:"jsoncs3"`
```
Comment (butonic) [inline services/sharing/pkg/config/config.go:115]:
```suggestion
	JSONCS3 PublicSharingJSONCS3Driver `yaml:"jsoncs3"`
	CS3     PublicSharingCS3Driver     `yaml:"cs3"`
```

File: ocis/pkg/command/migrate.go [hunk 2]
```diff
@@ -275,6 +275,13 @@ func revaPublicShareConfig(cfg *sharing.Config) map[string]interface{} {
 			"file":         cfg.PublicSharingDrivers.JSON.File,
 			"gateway_addr": cfg.Reva.Address,
 		},
+		"jsoncs3": map[string]interface{}{
+			"gateway_addr":        cfg.PublicSharingDrivers.JSONCS3.ProviderAddr,
```
Comment (butonic) [inline ocis/pkg/command/migrate.go:279]: doesn't that neet the real gateway?
```suggestion
			"gateway_addr":        cfg.Reva.Address,
```

File: services/sharing/pkg/revaconfig/config.go [hunk 3]
```diff
@@ -89,6 +89,13 @@ func SharingConfigFromStruct(cfg *config.Config) map[string]interface{} {
 							"service_user_idp":    cfg.PublicSharingDrivers.CS3.SystemUserIDP,
 							"machine_auth_apikey": cfg.PublicSharingDrivers.CS3.SystemUserAPIKey,
 						},
+						"jsoncs3": map[string]interface{}{
+							"gateway_addr":        cfg.PublicSharingDrivers.JSONCS3.ProviderAddr,
```
Comment (butonic) [inline services/sharing/pkg/revaconfig/config.go:93]:
```suggestion
							"gateway_addr":        cfg.Reva.Address,
```

**Review by butonic — APPROVED** [review]

**Comment by butonic** [comment]
and add a go.mod update?

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-6</strong>](https://drone.owncloud.com/owncloud/ocis/15346/50/1) failed. Further test are cancelled...

**Comment by micbar** [comment]
fails with 
```+ ocis/bin/ocis server
{"level":"error","service":"sharing","error":"unable to register services: rgrpc: grpc service publicshareprovider could not be started,: error: not found: driver not found: jsoncs3","time":"2022-09-09T11:59:53.011937951Z","message":"error starting the grpc server"}```

**Pull request merged**